### PR TITLE
moves code around

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -91,31 +87,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func (r *Resource) applyPatches(ctx context.Context, accessor metav1.Object, patches []Patch) error {
```
Comment (xh3b4sd) [inline create.go]: Everything started with creation. So everything was in this file. Now there is shared code for deletion as well. Thus this gets moved to a more common place.

File: resource.go [hunk 2]
```diff
@@ -93,3 +100,36 @@ func New(config Config) (*Resource, error) {
...
+	p := ensureSelfLink(accessor.GetSelfLink())
+
+	err = r.restClient.Patch(types.JSONPatchType).AbsPath(p).Body(b).Do().Error()
+	if errors.IsConflict(err) {
+		return microerror.Mask(err)
+	} else if errors.IsResourceExpired(err) {
+		return microerror.Mask(err)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
```
Comment (xh3b4sd) [inline resource.go]: Note the weirdness here with the error handling. This is for debugging and find out what kind of error we get when the patch fails due to the failed resource version test. Never happened so far. Just keeping for a while.

Comment (rossf7) [reply resource.go]: Cool thanks for the explanation.

**Review by rossf7 — APPROVED** [review]
LGTM

File: resource.go (hunk 2 above)
Comment (kopiczko) [reply resource.go]: I would leave it as it is even after debugging. It makes it clear what errors this function can produce.

**Pull request merged**